src/main/java/uk/co/ramp/covid/simulation/population/Adult.java

```diff
@@ -48,37 +48,58 @@ public void allocateCommunalPlace(Places p) {
             case TEACHER: {
                 School s = p.getRandomSchool();
                 setPrimaryPlace(s);
-                shifts = s.getShifts();
+                if (s == null) {
```

Comment:
Would it be possible to replace all these calls to getShifts() with something after the switch statement?  Perhaps along the lines of:
```
if (primaryPlace != null) {
    shifts = primaryPlace.getShifts();
}
```
(I guess I'm missing something here.)

Reply:
Yes, that works

Reply:
Thanks.  But I'm afraid that by changing it you just demonstrated that my suggestion was wrong!  It now looks like the calls to `setPrimaryPlace` should be moved, and it should be:
```
if (primaryPlace != null) {
    setPrimaryPlace(primaryPlace);
    shifts = primaryPlace.getShifts()
}
```
BUT I'm not sure that that is correct either, since `setPrimaryPlace` is setting `this.primaryPlace`, which could be made `protected` instead of `private`.

Sorry to be difficult.  I'll stop commenting about this now!

Reply:
No it's my fault. It does work as you suggest,